### PR TITLE
📝💅 Split trivial change log category into 3

### PR DESCRIPTION
**Pull request by webknjaz**
The new change note types are `packaging`, `contrib` and `misc`.
`packaging` is intended for the audience of downstream redistributors.
The `contrib` notes are meant to be documenting news affecting the
project contributors, their development, and processes.
Finally, `misc` is for things that don't fit anywhere but are still
desired to be documented for some reason.

This patch additionally extends the change log title meanings, making them more verbose and clearer to the
readers. Plus, it adds comment annotations in the Towncrier configuration.

### CODE REVIEW

**Comment by webknjaz** [comment]
FTR, some of the reasoning/justification is present in this external discussion: https://github.com/pypa/pip/issues/12555.

**Review by RonnyPfannschmidt — APPROVED** [review]

**Comment by RonnyPfannschmidt** [comment]
should this backport?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -66,8 +66,8 @@ repos:
     -   id: changelogs-rst
         name: changelog filenames
         language: fail
-        entry: 'changelog files must be named ####.(breaking|bugfix|deprecation|doc|feature|improvement|trivial|vendor).rst'
-        exclude: changelog/(\d+\.(breaking|bugfix|deprecation|doc|feature|improvement|trivial|vendor).rst|README.rst|_template.rst)
+        entry: 'changelog files must be named ####.(breaking|bugfix|contrib|deprecation|doc|feature|improvement|misc|packaging|vendor).rst'
+        exclude: changelog/(\d+\.(breaking|bugfix|contrib|deprecation|doc|feature|improvement|misc|packaging|vendor).rst|README.rst|_template.rst)
```
Comment (RonnyPfannschmidt) [inline .pre-commit-config.yaml:70]: this one is missing the .gitignore

**Review by RonnyPfannschmidt — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (webknjaz) [reply .pre-commit-config.yaml]: @RonnyPfannschmidt fixed! Marking this thread as resolved.

**Review by RonnyPfannschmidt — APPROVED** [review]
:heart:

**Comment by webknjaz** [comment]
FYI the preview for this PR's change note is rendered @ https://pytest--12501.org.readthedocs.build/en/12501/changelog.html#contributor-facing-changes.

**Comment by webknjaz** [comment]
Manual backport PR: #12506

**Pull request merged**